### ISSUE TITLE
Arcus.Templates.AzureFunctions.Http template correlation properties not sent to Application Insights

### ISSUE THREAD

**Issue by pim-simons** [arcus-azure/arcus.templates#393]
**Describe the bug**
When using the Arcus.Templates.AzureFunctions.Http template it returns the correlation properties (X-Transaction-ID, RequestId) in the response, but these properties are missing in the logging in Application Insights. 

**To Reproduce**
Create an Azure Function based on the Arcus.Templates.AzureFunctions.Http template and log some information to Application Insights.

**Expected behavior**
The Application Insights data to have the correlation properties. 

**Additional context**
- Version
Arcus.Templates - 0.6.0
Arcus.Observability - 2.2.1

**Comment by stijnmoreels** [comment]
@pim-simons , as discussed offline: not currently available for us to do in a stable manner as there's no `IServiceProvider` reference in the 'hook' injection points of Azure Functions. Our correlation model now requires such a reference to retrieve the correlation information of the incoming HTTP request.

**Comment by pim-simons** [comment]
As a workaround I did get the link between the logged information working by changing the TryHttpCorrelate a bit:
```
if (TryHttpCorrelate(out string errorMessage) == false)
{
	httpStatusCode = System.Net.HttpStatusCode.BadRequest;
	return BadRequest(errorMessage);
}
else
{
	var correlationInfo = _httpCorrelation.GetCorrelationInfo();
	telemetryContext.Add("Transaction", correlationInfo.TransactionId);
	telemetryContext.Add("OperationId", correlationInfo.OperationId);
	telemetryContext.Add("OperationParentId", correlationInfo.OperationParentId);
}
```

Now I can search on the correlation information in Application Insights, which is quite valuable and the link is made, only thing that is not right is the parent-child connection (in the image below 1 is the parent and 2 is the child):
![image](https://user-images.githubusercontent.com/32359437/125949618-32e055b9-17b5-4937-865e-8ca05b5d507b.png)

**Comment by pim-simons** [comment]
Some more info, according to this example https://docs.microsoft.com/en-us/azure/azure-monitor/app/correlation#example, the ID should be filled in as well. In my case this is always empty for dependancies and for the request it is n/a.
![image](https://user-images.githubusercontent.com/32359437/125999310-7af0173b-0cc6-4d72-a28d-e084306cfde0.png)

I tried filling several TelemetryContext fields (Id, DependencyId) but none seem to work.
Is there any way to fill the ID? And any clue why the request contains n/a?

**Comment by tomkerkhove** [comment]
Nope, we are going to add service-to-service correlation in one of our next iterations on observability - https://github.com/arcus-azure/arcus.observability/issues/78

**Comment by pim-simons** [comment]
I have a customer running some API's based on the Arcus.Templates.AzureFunctions.Http template in production, so if you need any data from app insights or an environment where we can test changes let me know.
